bgpd/bgp_vty.c

```diff
@@ -1448,8 +1448,21 @@ DEFUN_YANG(no_router_bgp,
 			struct bgp *tmp_bgp;
 
 			for (ALL_LIST_ELEMENTS_RO(bm->bgp, node, tmp_bgp)) {
-				if (tmp_bgp->inst_type
-				    == BGP_INSTANCE_TYPE_VRF) {
+				if (tmp_bgp->inst_type != BGP_INSTANCE_TYPE_VRF)
+					continue;
+				if (CHECK_FLAG(tmp_bgp->af_flags[AFI_IP][SAFI_UNICAST],
+					       BGP_CONFIG_MPLSVPN_TO_VRF_IMPORT) ||
+				    CHECK_FLAG(tmp_bgp->af_flags[AFI_IP6][SAFI_UNICAST],
+					       BGP_CONFIG_MPLSVPN_TO_VRF_IMPORT) ||
+				    CHECK_FLAG(tmp_bgp->af_flags[AFI_IP][SAFI_UNICAST],
+					       BGP_CONFIG_VRF_TO_MPLSVPN_EXPORT) ||
+				    CHECK_FLAG(tmp_bgp->af_flags[AFI_IP6][SAFI_UNICAST],
+					       BGP_CONFIG_VRF_TO_MPLSVPN_EXPORT) ||
+				    CHECK_FLAG(tmp_bgp->af_flags[AFI_L2VPN][SAFI_EVPN],
+					       BGP_L2VPN_EVPN_ADVERTISE_IPV4_UNICAST) ||
+				    CHECK_FLAG(tmp_bgp->af_flags[AFI_L2VPN][SAFI_EVPN],
+					       BGP_L2VPN_EVPN_ADVERTISE_IPV6_UNICAST) ||
+				    (tmp_bgp->vnihash && hashcount(tmp_bgp->vnihash))) {
```

Comment:
Is this check is necessary while you check already for L2VPN EVPN?

Reply:
actually, I improved it. the constraint i know is we do evpn rt5 on one vrf instance , and the main evpn instance is the bgp main one.
my answer is we check for the config evpn rt5 if we want to destroy the main bgp evpn instance